### PR TITLE
[choreolib] Fixed `done` logic

### PR DESCRIPTION
**Pull request by oh-yes-0-fps**
`done` is extensively document in its specific behavior and why it is the way it is.

The commit history looks weird here when un-squashed, this is built off of #881 with just reverting the commit in that where the change to `done` logic was reverted.
I'm sure there was a better way but idk how to do it

Requires #881 to be merged first

### CODE REVIEW

**Review by spacey-sooty — DISMISSED** [review]
Rebase

**Review by spacey-sooty — CHANGES_REQUESTED** [review]
I'd like a test to make sure this behaviour doesn't break

**Comment by oh-yes-0-fps** [comment]
I ran it in sim in my robot project and there were no regressions. I will be running in actual matches in abt 8 hrs

**Comment by spacey-sooty** [comment]
I mean like a unit test

File: choreolib/src/main/java/choreo/auto/AutoFactory.java [hunk 1]
```diff
@@ -83,6 +83,11 @@ public void poll() {}
         @Override
         public void reset() {}
 
+        @Override
+        public boolean isMostRecentTrajectory(AutoTrajectory trajectory) {
+          return false;
+        }
```
Comment (spacey-sooty) [inline choreolib/src/main/java/choreo/auto/AutoFactory.java:89]: Can you add a comment to explain this?

**Review by spacey-sooty — APPROVED** [review]

**Pull request merged**